docs/reference/v2_inference/README.md

```diff
@@ -0,0 +1,373 @@
+
+
+
+
+#### ModelInferResponse.InferOutputTensor.ParametersEntry
+
+
+
+
+| Field | Type | Description |
+| ----- | ---- | ----------- |
+| key | [string](#string) | N/A |
+| value | [InferParameter](#inferparameter) | N/A |
+
+
+
+
+#### ModelInferResponse.ParametersEntry
+
+
+
+
+| Field | Type | Description |
+| ----- | ---- | ----------- |
+| key | [string](#string) | N/A |
+| value | [InferParameter](#inferparameter) | N/A |
+
+
+
+
+#### ModelMetadataRequest
+
+
+
+
+| Field | Type | Description |
+| ----- | ---- | ----------- |
+| name | [string](#string) | The name of the model. |
+| version | [string](#string) | The version of the model to check for readiness. If not given the server will choose a version based on the model and internal policy. |
+
+
+
+
+#### ModelMetadataResponse
+
+
+
+
+| Field | Type | Description |
+| ----- | ---- | ----------- |
+| name | [string](#string) | The model name. |
+| versions | [repeated string](#string) | The versions of the model available on the server. |
+| platform | [string](#string) | The model's platform. See Platforms. |
+| inputs | [repeated ModelMetadataResponse.TensorMetadata](#modelmetadataresponse-tensormetadata) | The model's inputs. |
+| outputs | [repeated ModelMetadataResponse.TensorMetadata](#modelmetadataresponse-tensormetadata) | The model's outputs. |
+
+
+
+
+#### ModelMetadataResponse.TensorMetadata
+
+Metadata for a tensor.
+
+
+| Field | Type | Description |
+| ----- | ---- | ----------- |
+| name | [string](#string) | The tensor name. |
+| datatype | [string](#string) | The tensor data type. |
+| shape | [repeated int64](#int64) | The tensor shape. A variable-size dimension is represented by a -1 value. |
+
+
+
+
+#### ModelReadyRequest
+
+
+
+
+| Field | Type | Description |
+| ----- | ---- | ----------- |
+| name | [string](#string) | The name of the model to check for readiness. |
+| version | [string](#string) | The version of the model to check for readiness. If not given the server will choose a version based on the model and internal policy. |
+
+
+
+
+#### ModelReadyResponse
+
+
+
+
+| Field | Type | Description |
+| ----- | ---- | ----------- |
+| ready | [bool](#bool) | True if the model is ready, false if not ready. |
+
+
+
+
+#### ServerLiveRequest
+
+
+
+
+
+
+#### ServerLiveResponse
+
+
+
+
+| Field | Type | Description |
+| ----- | ---- | ----------- |
+| live | [bool](#bool) | True if the inference server is live, false if not live. |
+
+
+
+
+#### ServerMetadataRequest
+
+
+
+
+
+
+#### ServerMetadataResponse
+
+
+
+
+| Field | Type | Description |
+| ----- | ---- | ----------- |
+| name | [string](#string) | The server name. |
+| version | [string](#string) | The server version. |
+| extensions | [repeated string](#string) | The extensions supported by the server. |
+
+
+
+
+#### ServerReadyRequest
```

Comment:
@yuzisun this is missing?

Reply:
@andyi2it

Reply:
also there is no description under a few of these.. is that intentional?

Reply:
do we want to add something?

Reply:
This is generated from the proto schema which is empty struct
https://github.com/kserve/open-inference-protocol/blob/main/specification/protocol/open_inference_grpc.proto#L55